platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -235,8 +230,25 @@ public void deserialize(
         otherParentHash = in.readSerializable(false, Hash::new);
         timeCreated = in.readInstant();
         in.readInt(); // read serialized length
-        transactions = in.readSerializableArray(ConsensusTransactionImpl[]::new, maxTransactionCount, true);
-        checkUserTransactions();
+
+        final TransactionConfig transactionConfig = ConfigurationHolder.getConfigData(TransactionConfig.class);
+
+        transactions = in.readSerializableArray(
+                ConsensusTransactionImpl[]::new, transactionConfig.maxTransactionCountPerEvent(), true);
+
+        int totalTransactionSize = 0;
+        for (final Transaction transaction : transactions) {
+            if (!hasUserTransactions && !transaction.isSystem()) {
+                hasUserTransactions = true;
+            }
+
+            totalTransactionSize += transaction.getSerializedLength();
+        }
+
+        if (totalTransactionSize > transactionConfig.maxTransactionBytesPerEvent()) {
+            throw new IOException("Total size of transactions in this event %s exceeds the maximum allowed size"
+                    .formatted(totalTransactionSize));
+        }
```

Comment:
Can this be moved to a helper function? Might reduce probably of merge conflicts with @edward-swirldslabs's pending event changes.

Reply:
Given that my merge has to wait for HIP approval, I am expecting merge conflicts to accumulate.  I don't mind absorbing them.

Reply:
I split this logic off into a helper method (with an unfortunately awkward name)